shopwild-frontend/src/components/Cart/CartProduct/CartProduct.jsx

```diff
@@ -0,0 +1,133 @@
+import { useData } from "../../../context/data-context";
+
+import serverRequests from "../../../server/serverRequests";
+
+const CartProduct = (props) => {
+  const { dispatch } = useData();
+
+  const utilToast = (setToast, setToastMessage, message) => {
+    setToast(true);
+    setToastMessage(message);
+    setTimeout(() => {
+      setToast(false);
+    }, 2000);
+  };
+
+  const clickHandler = async (e) => {
+    if (e.target.textContent === "MOVE TO WISHLIST") {
+      const { error } = await serverRequests({
+        requestType: "post",
+        url: `${process.env.REACT_APP_BACKEND}/wishlist/607d92eee69f8b99745ef728`,
+        data: { products: [{ product: props.id, isWishlisted: true }] },
+      });
+
+      if (!error) {
+        dispatch({
+          type: "ADD_TO_WISHLIST",
+          payload: { product: { ...props }, isWishlisted: true },
+        });
+        utilToast(props.setToast, props.setToastMessage, "Moved to Wishlist!");
+      }
+    }
+
+    const { error } = await serverRequests({
+      requestType: "post",
+      url: `${process.env.REACT_APP_BACKEND}/cart/607d92eee69f8b99745ef728`,
+      data: { products: [{ product: props.id, quantity: 0 }] },
+    });
+
+    if (!error) {
+      dispatch({ type: "REMOVE_FROM_CART", payload: props.id });
+    }
+
+    if (e.target.textContent === "REMOVE") {
+      utilToast(props.setToast, props.setToastMessage, "Removed from Cart!");
+    }
+  };
+
+  const increaseQuantityHandler = async () => {
+    const { error } = await serverRequests({
+      requestType: "post",
+      url: `${process.env.REACT_APP_BACKEND}/cart/607d92eee69f8b99745ef728`,
+      data: {
+        products: [{ product: props.id, quantity: props.quantity + 1 }],
+      },
+    });
+    if (!error) {
+      dispatch({ type: "INCREASE_QUANTITY", payload: props.id });
+    }
+  };
+
+  const decreaseQuantityHandler = async () => {
+    if (props.quantity > 1) {
+      const { error } = await serverRequests({
+        requestType: "post",
+        url: `${process.env.REACT_APP_BACKEND}/cart/607d92eee69f8b99745ef728`,
```

Comment:
Im not sure how your backend is structured but im curious to know why you have added unique id after /cart/ will it be same to every user or planning to change after doing auth on backend

Reply:
that's user id, as of now only one user is there in DB. Once auth on backend is done, will refactor that.